File: skyspacefourtyfour_ui/src/services/CheckInService.js
import http from "../http-common"; 

class CheckInService {
  getAllCheckIns(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/checkIn/checkIns`, searchDTO);
  }

  get(checkInId) {
    return this.getRequest(`/checkIn/${checkInId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/checkIn?field=${matchData}`, null);
  }

  addCheckIn(data) {
    return http.post("/checkIn/addCheckIn", data);
  }

  update(data) {
  	return http.post("/checkIn/updateCheckIn", data);
  }
  
  uploadImage(data,checkInId) {
  	return http.postForm("/checkIn/uploadImage/"+checkInId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CheckInService();

File: src/main/java/com/skyspacefourtyfour/service/CheckInService.java
package com.skyspacefourtyfour.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefourtyfour.domain.CheckIn;
import com.skyspacefourtyfour.dto.CheckInDTO;
import com.skyspacefourtyfour.dto.CheckInSearchDTO;
import com.skyspacefourtyfour.dto.CheckInPageDTO;
import com.skyspacefourtyfour.dto.CheckInConvertCriteriaDTO;
import com.skyspacefourtyfour.service.GenericService;
import com.skyspacefourtyfour.dto.common.RequestDTO;
import com.skyspacefourtyfour.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CheckInService extends GenericService<CheckIn, Integer> {

	List<CheckIn> findAll();

	ResultDTO addCheckIn(CheckInDTO checkInDTO, RequestDTO requestDTO);

	ResultDTO updateCheckIn(CheckInDTO checkInDTO, RequestDTO requestDTO);

    Page<CheckIn> getAllCheckIns(Pageable pageable);

    Page<CheckIn> getAllCheckIns(Specification<CheckIn> spec, Pageable pageable);

	ResponseEntity<CheckInPageDTO> getCheckIns(CheckInSearchDTO checkInSearchDTO);
	
	List<CheckInDTO> convertCheckInsToCheckInDTOs(List<CheckIn> checkIns, CheckInConvertCriteriaDTO convertCriteria);

	CheckInDTO getCheckInDTOById(Integer checkInId);







}





